src/main/java/uk/gov/hmcts/reform/pip/data/management/service/PublicationService.java

```diff
@@ -37,20 +37,24 @@ public PublicationService(ArtefactRepository artefactRepository,
      * @return Returns the UUID of the artefact that was created.
      */
     public Artefact createPublication(Artefact artefact, String payload) {
+        boolean payloadValidAndAccepted = payloadExtractor.acceptAndValidate(payload);
+        if (payloadValidAndAccepted) {
+            Optional<Artefact> foundArtefact =  artefactRepository
+                .findBySourceArtefactIdAndProvenance(artefact.getSourceArtefactId(), artefact.getProvenance());
 
-        Optional<Artefact> foundArtefact =  artefactRepository
-            .findBySourceArtefactIdAndProvenance(artefact.getSourceArtefactId(), artefact.getProvenance());
+            foundArtefact.ifPresent(value -> artefact.setArtefactId(value.getArtefactId()));
 
-        foundArtefact.ifPresent(value -> artefact.setArtefactId(value.getArtefactId()));
+            String blobUrl = azureBlobService.createPayload(
+                artefact.getSourceArtefactId(),
+                artefact.getProvenance(),
+                payload);
 
-        String blobUrl = azureBlobService.createPayload(
-            artefact.getSourceArtefactId(),
-            artefact.getProvenance(),
-            payload);
+            artefact.setPayload(blobUrl);
+            artefact.setSearch(payloadExtractor.extractSearchTerms(payload));
 
-        artefact.setPayload(blobUrl);
-        artefact.setSearch(payloadExtractor.extractSearchTerms(payload));
-
-        return artefactRepository.save(artefact);
+            return artefactRepository.save(artefact);
+        } else {
+            return null;
```

Comment:
We should use Optionals rather than nulls where possible. This prevents null pointers further down the line :)

Reply:
we don't use Optional in the signature